### DISCUSSION TITLE
Handling tool opts for different UVM versions

### DISCUSSION THREAD

**Discussion by tudortimi** [svunit/svunit#201] [Ideas]
I think something using just command line options would be better than using environment variables.

My issue with #60 is that it leads down the slippery road of having to always keep up with tool switches. If a new UVM version comes out, we'd have to update `runSVUnit`. Also, it could be the case that not all simulators support the same UVM versions.

(Eventually, I'd like to decouple SVUnit from having to also compile the code, instead having it rely on the user's build tool of choice. This is a long way off, because of the reliance on generated code, but this will hopefully go away. The self-registering tests we have as an experimental feature are a step in that direction.)

For `xrun`, the user can always just add `-uvmhome`, so an extra switch is not really needed. I generally add a `-uvmhome CDNS-1.2` in `svunit.f` for my projects where I need that UVM version. For VCS it's probably not possible to add `-ntb_opts uvm-1.2` if `runSVUnit` already tries to add `-ntb_opts uvm`, hence #60. Instead of passing a version string to `--uvm`, I'm thinking of adding a new `--uvm-opts` switch (or something like that), which takes a list of options the tool needs to build UVM. If this switch is present, those tool opts will be used instead of the built-in ones `runSVUnit` already adds. This way, we don't have to hard-code too much tool specific stuff in the scripting. I'm also thinking about a `--no-uvm-opts` switch, which stops SVUnit from trying to add any switches for building UVM, because, for example, the user added them to their `svunit.f`, specified them on the command line, etc.

**Comment by edcote** [comment]
https://github.com/svunit/svunit/pull/204
